hal/halx86/legacy/bussupp.c

```diff
@@ -825,18 +825,27 @@ HalpDebugPciDumpBus(IN ULONG i,
     {
         /* Isolate the subclass name */
         ClassName += 6;
-        sprintf(LookupString, "\t%02x  ", PciData->SubClass);
+        ClassEnd = strstr(ClassName, "\nC ");
```

Comment:
What would happen if we are currently looking at the very last class listed in the class list, something like:
```
C  some_class
<tab>subclass
[END-OF-FILE]
```
?
ClassEnd would be NULL, does the rest of the code correctly handles this?

Reply:
Yes, it will be correctly handled. If there is no next class, we can't match subclass name from it by definition. And there is a check in the code for it. :slightly_smiling_face: